### DISCUSSION TITLE
Variable keeps getting removed from imputation

### DISCUSSION THREAD

**Discussion by studentaccount221** [amices/mice#666] [Getting started]
I am trying to impute some missing values in my data. 
I have a variable which represents the time between onset of symptoms and time of randomization (t_or). I also have a variable time from onset to start surgery (t_os) and a variable time from onset to end surgery (t_oe). t_or has no missing values while t_os and t_oe do. I need all these variables for my analysis. 

I am using the pmm method and I have added the following to my predictor matrix: 

```
pred_matrix["t_os", "t_or"] <-  1
pred_matrix["t_os", "t_oe"] <-  0
pred_matrix["t_oe", c("t_or", "t_os")] <-  1
```

Now, i keep getting warning messages that in order to predict either t_os or t_oe, MICE has removed t_or. 
However, the time period from onset to randomization is important to predict the time period from onset to start surgery, so I need this variable. 

I am very new to MICE so I am not sure how to solve this issue. What can i do to solve this?

**Comment by stefvanbuuren** [comment]
The idea behind your code seems sound. 

Could other variables in your data be linearly related to `t_os`? Try creating a small reprex.
